### PR TITLE
Move BucketKeyProvider and ProfileProvider out of UserResolver

### PR DESCRIPTION
**Pull request by msimko81**
The changes are due to the need for different behaviour of the method `UserResolver.GetBucketAsync` in the case of a call from Migration application.

### CODE REVIEW

File: WebApplication/Services/LoggedInUserBucketKeyProvider.cs [hunk 1]
```diff
@@ -0,0 +1,42 @@
+﻿using System.Threading.Tasks;
+using Autodesk.Forge.Core;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Options;
+using WebApplication.Utilities;
+
+namespace WebApplication.Services
+{
+    public class LoggedInUserBucketKeyProvider : IBucketKeyProvider
```
Comment (mivasi) [inline WebApplication/Services/LoggedInUserBucketKeyProvider.cs:9]: What is the point of having LoggedInUserBucketKeyProvider as a base class for MigrationBucketKeyProvider?

Comment (msimko81) [reply WebApplication/Services/LoggedInUserBucketKeyProvider.cs:9]: Shared method GetBucketPrefix. Moved to the separate provider.

File: WebApplication.Tests/MigrationBucketKeyProvider.cs [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+namespace WebApplication.Services
+{
+    public class MigrationBucketKeyProvider : IBucketKeyProvider
+    {
+        public string BucketKey { private get; set; }
+
+        public Task<string> GetBucketKeyAsync()
+        {
+            return new Task<string>(() => BucketKey);
```
Comment (erymski) [inline WebApplication.Tests/MigrationBucketKeyProvider.cs:11]: Would be better to use `Task.FromResult(BucketKey)`

File: WebApplication/Services/MigrationBucketKeyProvider.cs [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+namespace WebApplication.Services
+{
+    public class MigrationBucketKeyProvider : IBucketKeyProvider
+    {
+        public string BucketKey { private get; set; }
+
+        public Task<string> GetBucketKeyAsync()
+        {
+            return new Task<string>(() => BucketKey);
```
Comment (erymski) [inline WebApplication/Services/MigrationBucketKeyProvider.cs:14]: Same `Task.FromResult()`

File: WebApplication.Tests/MigrationBucketKeyProvider.cs (hunk 2 above)
Comment (msimko81) [reply WebApplication.Tests/MigrationBucketKeyProvider.cs]: This file was on a wrong location, I have deleted it.

**Review by hradilm — APPROVED** [review]

**Review by erymski — APPROVED** [review]

**Pull request merged**